### ISSUE TITLE
UTF-8 Support

### ISSUE THREAD

**Issue by shibumi** [jwilm/alacritty#715]
Hello,
UTF-8 Support seems to be a problem. The terminal is absolutly incompatible with projects like: https://github.com/bhilburn/powerlevel9k

**Comment by DarkDefender** [comment]
This terminal supports utf-8. So you will have to be a bit more specific about what is not working.

One of the current know problems is that alacritty does not support zero width chars. But other than that, most utf-8 codes should work.

**Comment by shibumi** [comment]
Hello thats what I am talking about:

![](https://paste.archlinux.de/nV7K9/)

**Comment by DarkDefender** [comment]
The font size doesn't seem to match with termite. Make sure that you have the same font settings (if possible).

Powerline symbols works for me in alacritty, so I'm guessing that on your end it doesn't pick the right font with the powerline stuff. This is probably because at that font size, it picks chars from a font that doesn't have the powerline symbols. So if you manage to get the same font size as in termite I'm guessing that it will solve it self

**Comment by sharpner** [comment]
To hijack this issue, I can reproduce a failing UTF-8 Icon when using the following keycode (both alacritty and terminal are using fura nerd patched font):

`printf "\xE2\x9C\x97"`

Expected (Mac Terminal):
<img width="289" alt="terminal" src="https://user-images.githubusercontent.com/574362/30443621-79fba5dc-9980-11e7-9729-332e8f70efeb.png">
Actual:
<img width="281" alt="alacritty" src="https://user-images.githubusercontent.com/574362/30443633-8462e756-9980-11e7-8a12-c86d895b2e8f.png">

Any pointers on what I might be doing wrong? :-)

Most other icons work fine so far...

**Comment by jwilm** [comment]
Sounds like an issue with fallback fonts on macOS

**Comment by L3rchal** [comment]
I have similar problem that if I open alacaritty on macOS then my fish prompt looks like : 
```
Welcome to fish, the friendly interactive shell
~ â ¯â ¯â ¯
```
However if I open another fish then the prompt looks fine
```
Welcome to fish, the friendly interactive shell
~ â ¯â ¯â ¯ fish
Welcome to fish, the friendly interactive shell
~ ❯❯❯

```

**Comment by sharpner** [comment]
thanks for the hint @jwilm I used this [patch](https://github.com/algesten/alacritty/commit/1ecd26ef0467b367d8c59522ca0c60004f7a6436.patch) for now :-)

**Comment by PSSGCSim** [comment]
I can confirm this happening on Linux (Debian Buster) and not only macOS as said in this thread.

**Comment by sodiumjoe** [comment]
I believe this was fixed by #956, at least for macos

**Comment by chrisduerr** [comment]
Should be fixed, other font configuration issues are probably covered by #957, closing this for now unless the issue is still present.
